mdata/cache/ccache_metric.go

```diff
@@ -19,6 +19,8 @@ type CCacheMetric struct {
 
 	// the list of chunk time stamps in ascending order
 	keys []uint32
+
+	RawMetric string
```

Comment:
as i have mentioned before, our codebase is very GC-heavy. we need to pay attention to how many pointers we keep active. and a string has a pointer. instead of keeping the entire RawMetric string for each CCacheMetric why not just keep 1 `uint8` which holds the number to be subtracted from the key to get the raw key? it'll go easier on GC and save us memory as well

Reply:
(we could also store the len of the raw string in a uint8 but then we're capped to a key len of 255 which some metrics will want to exceed. but a suffix can't be longer than 255 chars, we know that for sure)

Reply:
https://github.com/grafana/metrictank/pull/555/commits/b5de5f35733dc491c2842e8f1762a67ad7f98cd7